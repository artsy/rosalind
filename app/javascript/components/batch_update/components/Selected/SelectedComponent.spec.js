import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { mount } from 'enzyme'
import { SelectedComponent } from './SelectedComponent'

let props

beforeEach(() => {
  props = {
    name: 'Foo Name',
    onRemove: jest.fn(),
  }
})

it('renders correctly', () => {
  const rendered = renderer.create(<SelectedComponent {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('fires the remove handler on click', () => {
  const wrapper = mount(<SelectedComponent {...props} />)
  const mockClickEvent = { preventDefault: jest.fn() }
  wrapper.find('a').simulate('click', mockClickEvent)
  expect(props.onRemove.mock.calls.length).toEqual(1)
})
