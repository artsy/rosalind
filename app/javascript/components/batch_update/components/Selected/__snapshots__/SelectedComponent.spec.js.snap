// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  Foo Name
  <a
    className="remove"
    href="#"
    onClick={[Function]}
  >
    âœ•
  </a>
</div>
`;
